### PR TITLE
Policies and readme for a full front-channel OIDC flow

### PR DESCRIPTION
**Pull request by graemefoster**
I've had a few requests for APIm to provide a OIDC oauth-reverse proxy similar to App Service Authentication.
The main reason is that some people cannot allow unauthenticated traffic to hit their backend APIs.

This set of policies provides the necessary endpoints / session management required for these flows.

NB: I'm not a security expert. I've done my best to follow the secure principals involved in OIDC, but it would be awesome if anyone with a security lens could review and feedback.

### CODE REVIEW

File: examples/oauth-proxy/readme.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# OAuth Proxy APIm policies
```
Comment (tomkerkhove) [inline examples/oauth-proxy/readme.md:1]: Can you change all "APIm" references to Azure API Management please?

Comment (tomkerkhove) [inline examples/oauth-proxy/readme.md:1]: Can you add an entry in https://github.com/Azure/api-management-policy-snippets/blob/master/examples/README.md please? We list every policy scenario there so we can remove this README here and align with existing approach

Comment (graemefoster) [reply examples/oauth-proxy/readme.md]: Done

Comment (graemefoster) [reply examples/oauth-proxy/readme.md]: Done

**Review by tomkerkhove — APPROVED** [review]

**Pull request merged**